update for hamlib4

dl1igc opened the pull request:
With this changes this library is able to compile against libham4. Basic function works with FTDX3000.

dl1igc committed:
update for hamlib4

xylo04 commented:
Since this isn't backward compatible with hamlib 3.x, goHamlib will need a [v2 branch](https://go.dev/blog/v2-go-modules). I support doing this.

dh1tw commented:
Hey @dl1igc, thanks for your PR! Very much appreciated. I share the concerns with @xylo04 and I think we should investigate if there is a way to maintain the compatibility with Hamlib 3. We will have to compare what has exactly changed between Hamlib v3 and v4. Maybe we could maintain compatibility with a few additional if/else statements. A v2 branch, as suggested by @xylo04 should be IMHO the last option.

dl1igc commented:
Thanks for your comments @xylo04 and @dh1tw. I can understand your concerns and share them. If I am not wrong, we have two different compatibility levels. We could probably do the golang code compatible to both libham version over preprocessor directives (#ifdefs and so) but this does not mean that the compiled version will be compatible with both libham versions, I am at the moment not sure that this will work. At the end I think this situation would be even more confusing for end users. I'll give this some deeper thought over the weekend.

xylo04 committed:
Declare this as a published module supporting hamlib v4

When https://github.com/dh1tw/goHamlib/pull/12 gets resolved, this fork will be obsolete.

xylo04 committed:
Fix constant

dl1igc committed:
Merge pull request #1 from xylo04/master

Merge from xylo04

dl1igc committed:
module declaration

dl1igc committed:
dependencies
